### DISCUSSION TITLE
Possibility to register custom hooks

### DISCUSSION THREAD

**Discussion by misaon** [nuxt/nuxt#16866] [Questions] [answered]
### Describe the feature

https://github.com/nuxt/framework/discussions/7490

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://v3.nuxtjs.org/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/framework/discussions) and [issues](https://github.com/nuxt/framework/issues).

**Comment by danielroe** [comment, marked as answer]
This is actually possible now. You need to augment the RuntimeNuxtHooks interface from `#app`.

**Reply by misaon** [reply]
I still have problems, the `RuntimeNuxtHooks `augument does not work. Even the custom hook in `vueuse/schema-org` doesn't work (see https://github.com/vueuse/schema-org/blob/6b849a8a3f54c4e696725b004c3e5f7a9f876b71/packages/nuxt/src/module.ts#L24).

My implementation:
```ts
declare module 'nuxt' {
  export interface RuntimeNuxtHooks {
    'my:test:hook': (example: string) => void
  }
}
```

**Reply by danielroe** [reply]
You need to augment `#app`, not `nuxt`.

**Reply by misaon** [reply]
No change after switch to #app
![image](https://user-images.githubusercontent.com/15806356/190391592-0803af9b-600f-4586-8830-91141760ee0f.png)

**Reply by misaon** [reply]
When i remove my declaration:

![image](https://user-images.githubusercontent.com/15806356/190394009-eb11f581-cdce-431e-9677-8117bccdf6b5.png)

**Reply by danielroe** [reply]
![CleanShot 2022-09-15 at 13 36 10](https://user-images.githubusercontent.com/28706372/190405196-4de5f9d7-c74c-4f24-afa2-95d161db0bd0.png)

```ts
export default defineNuxtPlugin(nuxtApp => {
  nuxtApp.hook('')
})

declare module '#app' {
  interface RuntimeNuxtHooks {
    'custom:hook': () => void | Promise<void>
  }
}
```

**Reply by misaon** [reply]
Maybe Jetbrains IDE issue. Thanks for reply
